marklogic-client-api/src/test/java/com/marklogic/client/test/rows/VectorTest.java

```diff
@@ -0,0 +1,155 @@
+	}
+
+	@Test
+	void cosineSimilarity_InvalidVector() {
+		PlanBuilder.ModifyPlan plan =
+			op.fromView("vectors", "persons")
+				.bind(op.as("sampleVector", invalidVector))
+				.bind(op.as("summaryCosineSim", op.vec.cosineSimilarity(op.col("embedding"),op.col("sampleVector"))))
+				.select(op.col("name"), op.col("summary"), op.col("summaryCosineSim"))
+				.orderBy(op.desc(op.col("summaryCosineSim")))
+				.limit(5);
+		Exception exception = assertThrows(FailedRequestException.class, () -> resultRows(plan));
+		String actualMessage = exception.getMessage();
+		assertTrue(actualMessage.contains("Server Message: XDMP-ARGTYPE"));
+		assertTrue(actualMessage.contains("arg2 is not of type vec:vector"));
+	}
+
+	@Test
+	void dotProduct() {
+		PlanBuilder.ModifyPlan plan =
+			op.fromView("vectors", "persons")
+				.bind(op.as("sampleVector", sampleVector))
+				.bind(op.as("summaryCosineSim", op.vec.dotProduct(op.col("embedding"),op.col("sampleVector"))))
+				.select(op.col("name"), op.col("summary"), op.col("summaryCosineSim"))
+				.orderBy(op.desc(op.col("summaryCosineSim")))
+				.limit(5);
+		List<RowRecord> rows = resultRows(plan);
+		assertEquals(2, rows.size());
+		rows.forEach(row -> {
+			// Simply a sanity check that the Dot Product is valid.
+			Assertions.assertTrue(row.getDouble("summaryCosineSim") > 0);
+		});
+	}
+
+	@Test
+	void euclideanDistance() {
+		PlanBuilder.ModifyPlan plan =
+			op.fromView("vectors", "persons")
+				.bind(op.as("sampleVector", sampleVector))
+				.bind(op.as("euclideanDistance", op.vec.euclideanDistance(op.col("embedding"),op.col("sampleVector"))))
+				.select(op.col("name"), op.col("summary"), op.col("euclideanDistance"))
+				.orderBy(op.desc(op.col("euclideanDistance")))
+				.limit(5);
+		List<RowRecord> rows = resultRows(plan);
+		assertEquals(2, rows.size());
+		rows.forEach(row -> {
+			// Simply a sanity check that the Dot Product is valid.
+			Assertions.assertTrue(row.getDouble("euclideanDistance") > 0);
+		});
+	}
+
+	@Test
+	void helpers() {
+		PlanBuilder.ModifyPlan plan =
+			op.fromView("vectors", "persons")
+				.bind(op.as("sampleVector", sampleVector))
+				.bind(op.as("dimension", op.vec.dimension(op.col("sampleVector"))))
+				.bind(op.as("normalize", op.vec.normalize(op.col("sampleVector"))))
+				.bind(op.as("magnitude", op.vec.magnitude(op.col("sampleVector"))))
+				.bind(op.as("get", op.vec.get(op.col("sampleVector"), op.xs.integer(2))))
+				.bind(op.as("add", op.vec.add(op.col("embedding"),op.col("sampleVector"))))
+				.bind(op.as("subtract", op.vec.subtract(op.col("embedding"),op.col("sampleVector"))))
+				.bind(op.as("base64Encode", op.vec.base64Encode(op.col("sampleVector"))))
+				.bind(op.as("base64Decode", op.vec.base64Decode(op.col("base64Encode"))))
+				.bind(op.as("subVector", op.vec.subvector(op.col("sampleVector"), op.xs.integer(1), op.xs.integer(1))))
+//				There seems to be a server bug preventing vectorScore from working. See ticket MLE-15508
```

Comment:
Make this a "@Disabled" test instead (with just the stuff in it that doesn't work). That's much easier to find later vs commented out code. Arguably, we can even let it fail, but I think "Disabled" is good.